test/compiler/inline.jl

```diff
@@ -1675,31 +1675,28 @@ function oc_capture_oc(z)
 end
 @test fully_eliminated(oc_capture_oc, (Int,))
 
+# inlining with unmatched type parameters
 @eval struct OldVal{T}
-    x::T
```

Comment:
The previous definition was actually invalid, causing segfault when concretized.